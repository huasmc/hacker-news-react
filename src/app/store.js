import { configureStore } from "@reduxjs/toolkit";
import dashboardReducer from "../features/dashboard/DashboardSlice";

export const store = configureStore({
	reducer: {
		dashboard: dashboardReducer,
	},
});
