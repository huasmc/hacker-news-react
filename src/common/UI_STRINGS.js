export const UI_STRINGS = {
	GENERAL: {
		TITLE: "HACKER NEWS",
	},
};
