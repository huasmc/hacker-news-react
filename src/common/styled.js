import styled from "styled-components";

export const Row = styled.div`
	display: grid;
	grid-template-columns: repeat(12, 1fr);
	grid-gap: 1rem;
`;

export const Col = styled.div`
	display: flex;
	align-items: center;
	justify-content: center;
	padding-left: 0.5rem;
	padding-right: 0.5rem;
	grid-column: span ${(props) => props.span};
`;
