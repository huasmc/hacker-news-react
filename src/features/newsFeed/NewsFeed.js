import { memo } from "react";
import { Col, Row } from "../../common/styled";
import { UI_STRINGS } from "../../common/UI_STRINGS";
import NewsFeedItem from "./components/NewsFeedItem";
import NewsFeedPagination from "./components/NewsFeedPagination";
import {
	StyledNewsFeedContainerLeft,
	StyledNewsFeedContainerRight,
	StyledPaginationContainer,
} from "./components/style";

const NewsFeed = ({ news, page, setPage }) => {
	const selectPage = (page) => setPage(page);

	return (
		<div>
			<Row span={12} style={{ minHeight: "570px" }}>
				{Array.isArray(news) && news.length === 0 && (
					<Row>
						<h2>{UI_STRINGS.NO_VALID_NEWS}</h2>
					</Row>
				)}
				<StyledNewsFeedContainerLeft>
					<Col span={6}>
						{Array.isArray(news) &&
							news.slice(0, 4).map((newsHit) => {
								return (
									<NewsFeedItem key={newsHit.objectID} newsHit={newsHit} />
								);
							})}
					</Col>
				</StyledNewsFeedContainerLeft>
				<StyledNewsFeedContainerRight>
					<Col span={6}>
						{Array.isArray(news) &&
							news.slice(4, 8).map((newsHit) => {
								return (
									<NewsFeedItem key={newsHit.objectID} newsHit={newsHit} />
								);
							})}
					</Col>
				</StyledNewsFeedContainerRight>
			</Row>
			<StyledPaginationContainer>
				<NewsFeedPagination page={page} selectPage={selectPage} />
			</StyledPaginationContainer>
		</div>
	);
};

export default memo(NewsFeed);
