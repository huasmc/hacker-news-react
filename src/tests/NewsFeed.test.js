import React from "react";
import { render } from "@testing-library/react";
import NewsFeed from "../features/newsFeed/NewsFeed";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { UI_STRINGS } from "../common/UI_STRINGS";

const news = [
	{
		objectID: "1",
		author: "John Doe",
		story_title: "Some news title",
		story_url: "https://some-news-title.com",
		created_at: "2022-01-01T00:00:00.000Z",
	},
	{
		objectID: "2",
		author: "Jane Doe",
		story_title: "Another news title",
		story_url: "https://another-news-title.com",
		created_at: "2022-02-01T00:00:00.000Z",
	},
];

describe("NewsFeed component", () => {
	it("should render correctly with empty news array", () => {
		const { container, getByText } = render(
			<Provider store={store}>
				<NewsFeed news={[]} />
			</Provider>
		);
		expect(container).toMatchSnapshot();
		expect(getByText(UI_STRINGS.NO_VALID_NEWS)).toBeInTheDocument();
	});

	it("should render correctly with news array of length 2", () => {
		const { container, getAllByTestId } = render(
			<Provider store={store}>
				<NewsFeed news={news} />
			</Provider>
		);
		expect(container).toMatchSnapshot();
		expect(getAllByTestId("news-item")).toHaveLength(2);
	});
});
