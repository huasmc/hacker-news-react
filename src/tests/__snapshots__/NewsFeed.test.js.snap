// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsFeed component should render correctly with empty news array 1`] = `
<div>
  <div>
    <div
      class="sc-bcXHqe ROMsa"
      span="12"
    >
      <div
        class="sc-bcXHqe ROMsa"
      >
        <h2>
          This page has no valid news
        </h2>
      </div>
      <div
        class="sc-jrcTuL gWbcAX"
      >
        <div
          class="sc-gswNZR doXCWo"
          span="6"
        />
      </div>
      <div
        class="sc-kDvujY dyApoO"
      >
        <div
          class="sc-gswNZR doXCWo"
          span="6"
        />
      </div>
    </div>
    <div
      class="sc-pyfCe kWIher"
    >
      <div
        class="sc-iBYQkv fusTzu"
      >
        <span
          style="font-weight: 700; cursor: pointer;"
        >
          &lt;
        </span>
      </div>
      <div
        class="sc-iBYQkv fusTzu"
      >
        <span
          data-testid="next-page"
          style="font-weight: 700; cursor: pointer;"
        >
          &gt;
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`NewsFeed component should render correctly with news array of length 2 1`] = `
<div>
  <div>
    <div
      class="sc-bcXHqe ROMsa"
      span="12"
    >
      <div
        class="sc-jrcTuL gWbcAX"
      >
        <div
          class="sc-gswNZR doXCWo"
          span="6"
        >
          <div
            class="sc-eDvSVe inBQGM"
          >
            <div
              class="news-row"
              data-testid="news-item"
            >
              <div
                class="sc-jSUZER bfLHfq"
              >
                <table>
                  <tbody>
                    <tr>
                      <td>
                        <img
                          alt=""
                          src="clock.svg"
                        />
                      </td>
                      <td
                        class="news-timestamp"
                      >
                        <span
                          class="sc-dkrFOg jkYCxt"
                        >
                          NaN hour ago by author
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
                <span
                  class="sc-hLBbgP bLLQyC"
                >
                   
                  Some news title
                </span>
              </div>
              <div
                class="sc-gKPRtg EMaks"
              >
                <img
                  alt=""
                  class="favorite-button"
                  data-testid="heart-unfilled"
                  src="heart-unfilled.svg"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-eDvSVe inBQGM"
          >
            <div
              class="news-row"
              data-testid="news-item"
            >
              <div
                class="sc-jSUZER bfLHfq"
              >
                <table>
                  <tbody>
                    <tr>
                      <td>
                        <img
                          alt=""
                          src="clock.svg"
                        />
                      </td>
                      <td
                        class="news-timestamp"
                      >
                        <span
                          class="sc-dkrFOg jkYCxt"
                        >
                          NaN hour ago by author
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
                <span
                  class="sc-hLBbgP bLLQyC"
                >
                   
                  Another news title
                </span>
              </div>
              <div
                class="sc-gKPRtg EMaks"
              >
                <img
                  alt=""
                  class="favorite-button"
                  data-testid="heart-unfilled"
                  src="heart-unfilled.svg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-kDvujY dyApoO"
      >
        <div
          class="sc-gswNZR doXCWo"
          span="6"
        />
      </div>
    </div>
    <div
      class="sc-pyfCe kWIher"
    >
      <div
        class="sc-iBYQkv fusTzu"
      >
        <span
          style="font-weight: 700; cursor: pointer;"
        >
          &lt;
        </span>
      </div>
      <div
        class="sc-iBYQkv fusTzu"
      >
        <span
          data-testid="next-page"
          style="font-weight: 700; cursor: pointer;"
        >
          &gt;
        </span>
      </div>
    </div>
  </div>
</div>
`;
